docker/Dockerfile.ubuntu

```diff
@@ -0,0 +1,92 @@
+# This file aimed for developers and optimized for quick re-builds
+# Shouild be used by running `make localdocker`
+FROM ffmpeg-base:nvidia as builder
+
+FROM golang:1-stretch as builder2
```

Comment:
I'm interested in really learning how this Dockerfile works! To that end ...

Is `golang:1-stretch` another Docker imagine that's been built somewhere else?

Reply:
Yes, docker image can use other images as base layers , so `golang:1-stretch` is a public image available in dockerhub , which comes with golang installed.

this structure allows us to have smaller final image too, since you can have a large base where the binary is being compiled, then you'd create a fresh, much smaller layer that will be used at runtime, and just copy the binary from the previous layer. since most of the dependencies at compile won't be needed at runtime.

Reply:
Very cool! That makes a lot of sense. That certainly makes deployment much easier/faster. Thanks for the explanation Yahya!